cypress/integration/tagsAndDescription.spec.js

```diff
@@ -0,0 +1,116 @@
+describe('Tags and Description', () => {
+    context('Train Dialogs', () => {
```

Comment:
@pnwbrian @SDESkowronski 

Here is outline of tests I think are relevant.  An alternative test categorization is to update all the existing tests with the tags and description stuff integrated within them. I could see it both ways but thought it was clear we are testing tags and description feature when its separated like this.

How do we make this format of writing tests work in the manager?
It seemed like the current code is only allowing a test group and name, and requires repeating the group and name for each test. I think it should be arbitrary and up to the test creator to decide how many levels make sense for those tests.  All our unit tests are written this way also.

End result to look something like this:
![image](https://user-images.githubusercontent.com/2856501/52736356-88432f80-2f7e-11e9-852f-933c0538b8d3.png)

Reply:
I ran the `cypress:dev` command you added and tried to run a train related test to see why the First Inputs are failing but it is not working.  I don't know how it's possible but the top level URL is changing to `localhost:5050` instead of the cypress frame url. 

See: http://recordit.co/2rurbLKsz3

Why is this happening?

I even tried modifying the `testList` array and running `(RunTestsFromList)` and it still didn't work.

Reply:
Did you resolve this? I answered your question about this in teams last week. If you merge in from Master you will have the fix.

Reply:
Yes, I saw that comment. Tests are running now. The ones that seem to be related to train dialogs which this PR would affect are all passing:

![image](https://user-images.githubusercontent.com/2856501/53039999-347c8e80-3435-11e9-9db1-2db0d3e4238c.png)

Oddly enough the tests that are failing are for model creation:

> Error: the string "Exact Match 'TEXT' NOT Found" was thrown, throw an Error :)

Which is coming from `ExactMatch` failure in this:
```
export function SelectType(type)
{
  cy.Get('[data-testid="dropdown-action-type"]').Click()
  cy.Get('button.ms-Dropdown-item').ExactMatch(type).Click()
}
```

Which this PR shouldn't have anything effect on.

Reply:
I think more importantly is the question:
> How do we make this format of writing tests work in the manager?

Certainly out of scope for PR but it is a goal for the long term and something I want you to think about.

Reply:
As for the way you have organized the test cases...

There is a lot of merit in your proposal, specifically I can see that the test list clearly documents what is covered by the test scenarios. However, the realities of test automation will likely restrain and curtail your goals.

Setup of the test case takes a significant amount of time, which adds up when spread across many test cases. Along the way many other aspects of the UI under test are exercised as well, so in reality you are really testing many of the features all at once. If we created the tests you describe in the image above it would take approximately 7.5 minutes to run v.s. 3.5 minutes if those scenarios were all tested in 2 or 3 test cases.

Also there is the extra time it take to carefully craft all of the extra test cases to focus on a single specific feature and scenario...maintenance costs too.

The way I have organized the tests are to take advantage of this reality and I test and validate (nearly) every step I take towards the end goal. There are far fewer individual test cases, yet they all test a lot of features. 

This by the way is how the Cypress documentation says it should be done as well. I was happy to see that in their docs since this has been my issue with test automation that I had to maintain on other teams I was on. There were way too many tests to maintain because it was organized more like you are suggesting. There was a lot of duplicate effort. Although they did use common helper methods, still there was a lot more code created to test than was really necessary.

Reply:
> This by the way is how the Cypress documentation says it should be done as well.

Yes, I think there are good reasons for it.

> Setup of the test case takes a significant amount of time  ... it would take approximately 7.5 minutes

What contributes to this time?  I thought in most cases we import a model which means there isn't any setup since it's pre-configured. I thought after training completes it would be ready to be tested?
If importing is very slow we could get by with creating a new model for this test since it only needs 1 action to select. From there it could be same train dialog for the remainder.  We don't have to re-import or create a model for each of those lines only once.

> Along the way many other aspects of the UI under test
> I test and validate (nearly) every step I take towards the end goal

The goal for these tests to verify tags and description. If we tests everything all the time this seems like it would take a long time and make the tests brittle. They could break due to unrelated changes which mislead people.  I would think the test should be isolated and specific so we know where the problem is when it occurs.  E.g. If I write a test for tags then later change how the tags work I would expect that test and only that to fail so I know which one to fix. It would not affect tests related to model creation, entities, or settings and other features.

> There were way too many tests to maintain
> There was a lot of duplicate effort.

I think it's about maximizing quality of product and developer productivity rather than minimizing test count.  It doesn't matter if there are 100's of tests if I only have to fix 1 that I broke and look in one location.  

I don't understand how there is duplicate effort.  I would think this categorization of tests means less duplicated effort because it doesn't spend time finding and asserting things that have already been asserted in previous tests.  

> I can see that the test list clearly documents what is covered by the test scenarios

Yes, that is very important and something current tests are missing.  Without looking at how the tests are implemented someone should be able to see what it's testing and asserting. Also when a test fails it will show an 'X' next to the tests that failed instead of you having to log every error and recreate a reporting system.

Example we just came across:
![image](https://user-images.githubusercontent.com/2856501/53050944-6bf83480-344f-11e9-98e0-ce3922bc80bc.png)

It only gives you the test name, not the specific failure. I would expect this to tell something like "should select the action type TEXT". If someone doesn't know where "TEXT" is written in the code or know it's a type of action they would not know where to look for this message. It could seemingly come from any where in the test.

Reply:
With regards to this issue of:
> Error: the string "Exact Match 'TEXT' NOT Found" was thrown, throw an Error :)

This is due to the ELECTRON browser being used by CircleCI runs...the .innerText() function off of the elements has a line feed character in it when the Chrome browser does not.